docs/reference/triggering.rst

```diff
@@ -0,0 +1,88 @@
+.. _trajectory_scan_triggering:
+
+Trajectory Scan Triggering
+==========================
+
+PMAC-based Triggering
+---------------------
+
+This is intended as a description of how trigger signals are used and generated
+for a trajectory scan. It uses the DLS trajectory scan motion program for PMAC
+and Power PMAC. An example of this in use can be found in the `pmac_tutorial`.
+
+There are 3 GPIO signals used to provide triggering information for detectors.
+All of these signals are considered to trigger on a rising edge. They are all
+fed into a PandA which uses them to determine when to average motor positions
+and when to trigger the detector(s). The signals are :
+
+* Live signal: Indicates the start of a frame
+* Dead signal: Indicates the start of a period where no frames are acquired
+* Centre signal: Indicates that we are in the middle of the current frame
+
+When a series of frames are taken back to back, the live signal triggers for
+the start of each. The dead signal triggers at the end of the last frame.
+
+When Malcolm uploads lists of PVT points to the trajectory scan motion program
+it also uploads a User Subroutine to be called as each point is reached. The
+first 8 of these represent all combinations of the 3 GPIO Signals as follows:
+
+====== ====== ====== ======
+User   GPIO 0 GPIO 1 GPIO 2
+Sub    Live   Dead   Centre
+====== ====== ====== ======
+1      0      0      1
+2      0      1      0
+3      0      1      1
+4      1      0      0
+5      1      0      1
+6      1      1      0
+7      1      1      1
+8      0      0      0
+====== ====== ====== ======
+
+The diagram below shows an example of a snake scan with 6 data points and the
+corresponding trigger signals.
+
+The graph at the bottom shows the states of the GPIO lines as the PVT scan
+progresses through its points.
+
+Note that the acquisition time is shorter than the live time. This trigger
+signal for the detector is controlled by the PandA. At present it pulses after
+the transition of the live signal, delayed by half the readout time. The end of
+acquisition is determined by the settings on the detector and this would be set
+to live time - readout time.
```

Comment:
There is nothing automatic that does this at the moment, although you can hard code it with a pulse Block.
When we roll out the pulse trigger part then it will do that automatically, and make the length of the trigger the same length as the exposure time

Reply:
After speaking to Giles about this, as far as I can tell the paragraph does correctly describe the trigger signal behaviour for current systems at Diamond. Do you have an example where we don't handle the readout time in such a manner?

Even if it is only a matter of convention, I still think this documentation should reflect current usage. There is a lot of flexibility when configuring Malcolm and PandA, and detailing it here could complicate things.

Reply:
Tom has suggested (in person) that this readout time delay is typically ignored for the majority of detectors.
I will amend the documentation to reflect this.